### PR TITLE
Get data branch

### PR DESCRIPTION
**Pull request by thomas-spriggs-codes**
Deleted a lot of Michael's code regarding date strings and saving/loading files, the api is super quick to access so I figured it was worth just downloading it all at once, every time. 

And added the feature where get_api takes the argument, n_days, specifying how many of the last n_days you want the data for.

Output of the code is still the same as it was for Michael's api bit of the old code. 

Ignore how many commits ahead of master I am, the old code was saving files all the time and i didn't realise so took a few commits to delete them all :)

### CODE REVIEW

**Review by TonichaC — APPROVED** [review]
Looks good

**Comment by mrknorman** [comment]
Well, I did that as you only get a certain number of downloads per day with the free API, so I figured it might be best if we weren't using it every time we tested.

File: get_data/get_data.py [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+from statsmodels.tsa.arima_model import ARIMA
+import pandas as pd
+import os
+
+from pandas.plotting import register_matplotlib_converters
+
+
+def call_api(n_days):
+    """
+	Calls the api using Michael's key, and returns the last n_days worth of WTI oil price data
```
Comment (hstewart93) [inline get_data/get_data.py:9]: 👍 yay for doc strings!

**Comment by thomas-spriggs-codes** [comment]
Ah, this i didnt know. So providing we dont mind having a few random files floating about I can bring back the old code? or we can run this once, and just use the result instead of constantly needing it to be up to date. Ill poll the slack?

File: get_data/get_data.py [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+	"""
+    # Set quandl API key
+    quandl.ApiConfig.api_key = "NszGhwY_Qh8Ubj1BWhVt"  # change this to a global constant, maybe even use the team key
+
+    # Get data from the quandl api, then only return the number of days asked for
+    data = quandl.get("EIA/PET_RWTC_D", returns="numpy")
+    return data[-n_days:]
```
Comment (hstewart93) [inline get_data/get_data.py:16]: This is a rather elegant solution 💯

File: get_data/get_data.py [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+import quandl
+from datetime import datetime
+from statsmodels.tsa.arima_model import ARIMA
+import pandas as pd
+import os
+
+from pandas.plotting import register_matplotlib_converters
+
```
Comment (hstewart93) [inline get_data/get_data.py:5]: Could you remove the unused imports here please? think you only need quandl?

**Review by hstewart93 — CHANGES_REQUESTED** [review]

**Comment by mrknorman** [comment]
Might be alright as it is a few hundred calls a day, just thought with all of us working on it might eventually reach that. But we could make it so it automatically deletes old files, or stores them in a folder? Either one.

**Comment by hstewart93** [comment]
> Well, I did that as you only get a certain number of downloads per day with the free API, so I figured it might be best if we weren't using it every time we tested.

This is a valid point Michael, obviously when running this for real we will want the most up to date data possible - maybe we can add an optional argument to use stored data for testing?

File: get_data/get_data.py (hunk 3 above)
Comment (thomas-spriggs-codes) [reply get_data/get_data.py:5]: yeah will do :)

**Review by hstewart93 — APPROVED** [review]

**Review by thomas-spriggs-codes — COMMENTED** [review]
I've pushed a new version to the same branch that has added your changes hattie

**Pull request merged**